src/blob.h

```diff
@@ -25,19 +23,30 @@ class Blob : public Napi::ObjectWrap<Blob> {
       // TODO(kelvinhammond): avoid copying content somehow in certain scenarios
       // if possible Maybe use a reference object??? What is the lifecycle of
       // that and what if it's changed?
+
+      size_t size = 0;
+      std::shared_ptr<char> data = nullptr;
       if (info[0].IsArrayBuffer()) {  // handle ArrayBuffer
         auto buf = info[0].As<Napi::ArrayBuffer>();
-        auto refcontent_ =
-            std::string(static_cast<char *>(buf.Data()), buf.ByteLength());
+        size = buf.ByteLength();
+        data = std::shared_ptr<char>(new char[size]);
+        memset(data.get(), 0, size);
```

Comment:
Is the `memset` really necassary as we will copy the data just after ?

Reply:
Not really, I just like to zero memory before use. I'll update it.

Reply:
Fixed in updated code